### PR TITLE
Python: Allow source dists to install correctly

### PR DESCRIPTION
**Pull request by paulmelnikow**
When a source distribution is installed, PEP 517 will cause pybind11 to be installed in a temporary location at build time.

https://martin-thoma.com/pyproject-toml/

Unfortunately when pybind11 is temporarily installled, it infers the wrong include path. This includes a workaround. Hopefully when that issue is fixed the workaround can be removed:

https://github.com/pybind/pybind11/issues/1067

I’ve also posted the workaround here:

https://github.com/pybind/python_example/issues/45

I've confirmed that this works with cibuildwheel in Azure Pipelines, which is a great way to produce wheels for distribution. I can contribute the config change in a separate PR.

### CODE REVIEW

**Comment by syoyo** [comment]
Awesome!

@paulmelnikow is it ready to merge this PR? or better to wait a fix for pybind11 for example this issue? >  https://github.com/pybind/python_example/issues/45

File: python/setup.py [hunk 1]
```diff
@@ -1,28 +1,80 @@
...
+                    "..",
+                    "include",
+                    os.path.basename(interpreter_include_path),
+                )
+            )
+        else:
+            return interpreter_include_path
+
 
 # `tiny_obj_loader.cc` contains implementation of tiny_obj_loader.
```
[the remark this answers is not in the document]
Comment (syoyo) [reply python/setup.py]: I see. Yes, please go ahead to add Black format check in CI. FYI I had an experiment to use yapf for another Python programs and didn't know `Black`. `Black` seems new and better than yapf.

**Comment by paulmelnikow** [comment]
I'd say it's ready to merge.

Since their generated include path is still being passed to `setuptools.Extension`, the sdists created with this script should be forward compatible if they fix the include path. Of course the wheels will keep working too, as they don't run this bit of code. When pybind/pybind11#1067 is fixed I'm happy to remove the workaround to tidy this up.

I'm eager to keep this moving forward – and especially to start shipping all the platform wheels!

File: python/setup.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (paulmelnikow) [reply python/setup.py]: Okay, I will do. I'll add that in a follow-on PR.

**Comment by syoyo** [comment]
> I'd say it's ready to merge.

Thanks! merged!

**Pull request merged**